### DISCUSSION TITLE
Selected strokes get deselected when Dialog option is clicked

### DISCUSSION THREAD

**Discussion by MiltonBalaOfficial** [xournalpp/xournalpp#6077] [Q&A]
When a option of a dialog is clicked, if the pen is selected then a small stroke is added on the page though not a big problem. Problem arises when selection tool is active and some strokes are selected. Because when the dialog option is clicked the selected strokes get deselected and no action can be done on those strokes through the dialog. But if the dialog option is over the selection then after clicking the selection remains as it is as the click is on the selection. What should I do to do some actions on the selected strokes through the dialog? I am giving a simple action plugin, If the action can be done on a selected stroke I may able to implement the method in my actual work. Test it when the dialog option is not over the selection. 

 ``function initUi()
    app.registerUi({
        ["menu"] = "Delete Through Dialog",
        ["callback"] = "openDialog",
    });
  end
  function openDialog()
    app.openDialog(
        "Do you want to Delete?", 
        {"YES", "NO"}, 
        "dialogCallback" 
    ) 
end

function dialogCallback(result)
  if result == 2 then
    return -- When "Cancel" option is clicked, it closes the dialog
  elseif result == 1 then
    app.uiAction({action = "ACTION_DELETE"})
  end
end
--I think when the dialog option is clicked the strokes get deselected before performing the action.``

**Comment by bhennion** [comment]
> When a option of a dialog is clicked, if the pen is selected then a small stroke is added on the page

This shouldn't be. What version are you running, on what OS and what's the GTK version (you can find this in the About dialog)?

**Reply by MiltonBalaOfficial** [reply]
![image](https://github.com/user-attachments/assets/065af362-c858-48e5-a806-2648dc4c64d9)
![image](https://github.com/user-attachments/assets/de11ff31-e27c-43f6-ace7-705468cf349d)

**Reply by MiltonBalaOfficial** [reply]
I thought that the problem might be due to my stylus. But I tested on linux, no dot!
I tested on windows. Some observations:
not only plugin options, cancel button of about dialog also and preferecce dialog also same problem but ok option of preference dialog doesn't produce dot

**Reply by MiltonBalaOfficial** [reply]
I checked it with mouse, no problem, no dot selection remains and plugin working properly. But what may be the problem with the stylus on windows?

**Reply by MiltonBalaOfficial** [reply]
I have tried two different writing  pad, same result: dot appears!

**Reply by bhennion** [reply]
You can open a bug ticket with that information, but it looks like a GTK bug and I'm not sure we can do anything about it.
If we can produce a minimal working example showing the bug, it can be reported to GTK
